### PR TITLE
Don't rely on gcc-specific features

### PR DESCRIPTION
**Pull request by lithp**
`#ifdef` away reliance on support for some features which are not supported by Visual Studio.

### CODE REVIEW

File: src/backend/distributed/worker/worker_partition_protocol.c [hunk 1]
```diff
@@ -614,6 +615,26 @@ CitusCreateDirectory(StringInfo directoryName)
 }
 
 
+#ifdef WIN32
+static bool
+FileIsLink(char *filename, struct stat filestat)
+{
+	DWORD attributes = GetFileAttributes(filename);
+	return attributes & FILE_ATTRIBUTE_REPARSE_POINT;
```
Comment (pykello) [inline src/backend/distributed/worker/worker_partition_protocol.c]: Why not use `pgwin32_is_junction()` instead?

File: src/backend/distributed/worker/worker_partition_protocol.c [hunk 2]
```diff
@@ -614,6 +615,26 @@ CitusCreateDirectory(StringInfo directoryName)
...
+{
+	return S_ISLNK(filestat.st_mode);
+}
+
+
+#endif
+
+
 /*
  * CitusRemoveDirectory first checks if the given directory exists. If it does, the
```
Comment (pykello) [inline src/backend/distributed/worker/worker_partition_protocol.c]: I think we should delete this function and use `rmtree()` instead. They seem to do the same thing.

Comment (lithp) [reply src/backend/distributed/worker/worker_partition_protocol.c]: Added some comments to your PR. I think that's a reasonable aim for a future PR but don't know if it belongs in this one.

File: src/backend/distributed/worker/worker_partition_protocol.c (hunk 1 above)
Comment (lithp) [reply src/backend/distributed/worker/worker_partition_protocol.c]: Nice catch. How did you find it?

Comment (pykello) [reply src/backend/distributed/worker/worker_partition_protocol.c]: I usually grep for functions/constants we use to check if postgres already uses them, and if yes, how :)

File: src/backend/distributed/worker/worker_partition_protocol.c (hunk 2 above)
Comment (pykello) [reply src/backend/distributed/worker/worker_partition_protocol.c]: 👍

**Pull request merged**